python/ray/autoscaler/autoscaler.py

```diff
@@ -429,7 +429,8 @@ def spawn_updater(self, node_id, init_commands, ray_start_commands,
             process_runner=self.process_runner,
             use_internal_ip=True,
             docker_config=self.config.get("docker"),
-            node_resources=node_resources)
+            node_resources=node_resources,
+            is_head_node=False)
         updater.start()
```

Comment:
Can you make sure anywhere NodeUpdaterThread is called, the appropriate `is_head_node` is invoked?

There's one in this class that doesn't seem to have it. 

Does it make sense to make `is_head_node` a mandatory field?

Reply:
Sure--I can make it mandatory!